project/SparkBuild.scala

```diff
@@ -240,7 +240,8 @@ object SparkBuild extends PomBuild {
 
     javacOptions in Compile ++= Seq(
       "-encoding", "UTF-8",
-      "-source", javacJVMVersion.value
+      "-source", javacJVMVersion.value,
+      "-Xlint:unchecked"
```

Comment:
This is enabling the unchecked checks, not suppressing them. I don't think you can enable the checks in source code.  And if I read things correctly, it's enabled in the maven build (`-Xlint:all,-serial,-path`).

I can remove the flag if it really bothers people, though (I didn't notice much extra noise from enabling it, and it helped me spot these occurrences in the code I'm adding.)